osu.Game/Overlays/Options/SliderOption.cs

```diff
@@ -1,53 +1,100 @@
-﻿using System;
+﻿using System;
+using OpenTK;
 using OpenTK.Graphics;
 using osu.Framework.Configuration;
 using osu.Framework.Graphics;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Primitives;
 using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
 using osu.Framework.Graphics.UserInterface;
 
-namespace osu.Game.Overlays.Options
-{
-    public class SliderOption<T> : FlowContainer where T : struct,
-        IComparable, IFormattable, IConvertible, IComparable<T>, IEquatable<T>
-    {
-        private SliderBar<T> slider;
-        private SpriteText text;
-    
-        public string LabelText
-        {
-            get { return text.Text; }
-            set
-            {
-                text.Text = value;
-                text.Alpha = string.IsNullOrEmpty(value) ? 0 : 1;
-            }
-        }
-        
-        public BindableNumber<T> Bindable
-        {
-            get { return slider.Bindable; }
-            set { slider.Bindable = value; }
-        }
-
-        public SliderOption()
-        {
-            Direction = FlowDirection.VerticalOnly;
-            RelativeSizeAxes = Axes.X;
-            AutoSizeAxes = Axes.Y;
-            Children = new Drawable[]
-            {
-                text = new SpriteText { Alpha = 0 },
-                slider = new SliderBar<T>
-                {
-                    Margin = new MarginPadding { Top = 5 },
-                    Height = 10,
-                    RelativeSizeAxes = Axes.X,
-                    Color = Color4.White,
-                    SelectionColor = new Color4(255, 102, 170, 255),
-                }
-            };
-        }
-    }
+namespace osu.Game.Overlays.Options
+{
+    public class SliderOption<T> : FlowContainer where T : struct,
+        IComparable, IFormattable, IConvertible, IComparable<T>, IEquatable<T>
+    {
+        private SliderBar<T> slider;
+        private SpriteText text;
+    
+        public string LabelText
+        {
+            get { return text.Text; }
+            set
+            {
+                text.Text = value;
+                text.Alpha = string.IsNullOrEmpty(value) ? 0 : 1;
+            }
+        }
+        
+        public BindableNumber<T> Bindable
+        {
+            get { return slider.Bindable; }
+            set { slider.Bindable = value; }
+        }
+
+        public SliderOption()
+        {
+            Direction = FlowDirection.VerticalOnly;
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+            Children = new Drawable[]
+            {
+                text = new SpriteText { Alpha = 0 },
+                slider = new OsuSliderBar<T>
+                {
+                    Margin = new MarginPadding { Top = 5 },
+                    RelativeSizeAxes = Axes.X,
+                }
+            };
+        }
+
+        private class OsuSliderBar<U> : SliderBar<U> where U : struct,

```

Comment:
i realy dislike this formatting. why not move the whole `where U...` to a new line so it's obvious that everything following it is related to U?